docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,415 @@
+
+[[create-client-implementation]]
+=== Create client implementation
+
+Finally, here is the client code to call the gRPC service deployed in Payara Server:
+
+[source, Java]
+----
+public class GrpcClient {
+    
+    private static final Logger LOGGER = Logger.getLogger(GrpcClient.class.getName());
+    private final PayaraServiceGrpc.PayaraServiceStub stub; //reference to the stub service implementation class
+    private CountDownLatch latch;
+    private AtomicReference<Throwable> error;
+
+    public static void main(String[] args) throws InterruptedException, MalformedURLException, URISyntaxException {
+        URL myURL = new URL("http://localhost:8080/fish.payara.samples.grpc.PayaraService"); // URL for the deployed gRPC service
+        final GrpcClient client = new GrpcClient(myURL); // creating client
+        client.communicate(); // call service
+    }
+
+    public GrpcClient(URL url) throws URISyntaxException {
+        final Channel channel = ManagedChannelBuilder.forAddress(url.getHost(), url.getPort())
+                .usePlaintext().build(); //creating channel to start communication to the service
+        this.stub = PayaraServiceGrpc.newStub(channel); //creating stub from the channel reference
+        this.error = new AtomicReference<>(null);
+    }
+
+    public void communicate() throws InterruptedException {
+        latch = new CountDownLatch(1); //this is to wait until the communication finish with the current thread
+        stub.communicate(request("Hello World"), new ResponseObserver()); //calling service and adding a ResponseObserver to process response
+        latch.await(20, TimeUnit.SECONDS); //timeout to wait response
+    }
+
+    public Throwable getError() {
+        return error.get();
+    }
+
+    private final class ResponseObserver implements StreamObserver<PayaraResp> {
+
+        @Override
+        public void onNext(PayaraResp response) { //to process the service response
+            LOGGER.log(Level.INFO, "Response received: \"{0}\".", response.getMessage()); // printing the response from the service
+        }
+
+        @Override
+        public void onError(Throwable t) { //method to process errors
+            LOGGER.log(Level.SEVERE, "Error received", t);
+            error.set(t);
+            latch.countDown();
+        }
+
+        @Override
+        public void onCompleted() {
+            latch.countDown(); //indicating that the communication complete for the current thread
+        }
+
+    }
+
+    private static final PayaraReq request(String message) {
+        return PayaraReq.newBuilder().setMessage(message).build(); //creating request with an String message
+    }
+}
+----
+
+[[execution-of-helloworld-application]]
+=== Execution of the "HelloWorld" application
+
+To execute the client application, build the project and run the following command:
+
+[source,shell]
+----
+${client app} mvn compile exec:java -Dexec.mainClass="fish.payara.samples.grpc.GrpcClient"
```

Comment:
I had this error in client:
```
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ payara-client ---
mar 11, 2022 10:49:21 AM fish.payara.samples.grpc.GrpcClient$ResponseObserver onError
GRAVE: Error received
io.grpc.StatusRuntimeException: CANCELLED
        at io.grpc.Status.asRuntimeException(Status.java:535)
        at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:479)
        at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:463)
        at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:427)
        at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:460)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

[WARNING] thread Thread[grpc-nio-worker-ELG-1-1,5,fish.payara.samples.grpc.GrpcClient] was interrupted but is still alive after waiting at least 14999msecs
[WARNING] thread Thread[grpc-nio-worker-ELG-1-1,5,fish.payara.samples.grpc.GrpcClient] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-0,5,fish.payara.samples.grpc.GrpcClient] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-1,5,fish.payara.samples.grpc.GrpcClient] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-2,5,fish.payara.samples.grpc.GrpcClient] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-2,5,fish.payara.samples.grpc.GrpcClient] will linger despite being asked to die via interruption
[WARNING] NOTE: 5 thread(s) did not finish despite being asked to  via interruption. This is not a problem with exec:java, it is a problem with the running code. Although not serious, it should be remedied.
[WARNING] Couldn't destroy threadgroup org.codehaus.mojo.exec.ExecJavaMojo$IsolatedThreadGroup[name=fish.payara.samples.grpc.GrpcClient,maxpri=10]
java.lang.IllegalThreadStateException
    at java.lang.ThreadGroup.destroy (ThreadGroup.java:778)
    at org.codehaus.mojo.exec.ExecJavaMojo.execute (ExecJavaMojo.java:293)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Reply:
Can you double check if the grpc module was added to the server and also the web application was deployed?